app/src/main/java/com/compscieddy/meetinthemiddle/holder/ChatHolder.java

```diff
@@ -1,68 +1,69 @@
 
-import android.graphics.PorterDuff;
 import android.graphics.drawable.GradientDrawable;
-import android.graphics.drawable.RotateDrawable;
 import android.support.v4.content.ContextCompat;
 import android.support.v7.widget.RecyclerView;
 import android.view.Gravity;
 import android.view.View;
-import android.widget.FrameLayout;
 import android.widget.LinearLayout;
-import android.widget.RelativeLayout;
 import android.widget.TextView;
 
 import com.compscieddy.meetinthemiddle.R;
 
+import java.util.Date;
+
 import butterknife.Bind;
 import butterknife.ButterKnife;
 
 /**
  * Created by elee on 7/6/16.
  */
-
 public class ChatHolder extends RecyclerView.ViewHolder {
 
-  @Bind(R.id.left_arrow) FrameLayout leftArrow;
-  @Bind(R.id.right_arrow) FrameLayout rightArrow;
-  @Bind(R.id.message_container) RelativeLayout messageContainer;
+  @Bind(R.id.message_container) LinearLayout messageContainer;
   @Bind(R.id.message_box) LinearLayout messageBox;
+  @Bind(R.id.name_text) TextView userName;
+  @Bind(R.id.time_stamp_text) TextView timestampText;
   View rootView;
+  TextView messageText;
+
 
   public ChatHolder(View itemView) {
     super(itemView);
     rootView = itemView;
     ButterKnife.bind(ChatHolder.this, rootView);
+    messageText = (TextView) rootView.findViewById(R.id.message_text);
   }
 
   public void setIsSender(boolean isSender) {
-    int color;
+    int bg_color;
+    int text_color;
     if (isSender) {
-      color = ContextCompat.getColor(rootView.getContext(), R.color.group_chat_background_color);
-      leftArrow.setVisibility(View.GONE);
-      rightArrow.setVisibility(View.VISIBLE);
+      bg_color = ContextCompat.getColor(rootView.getContext(), R.color.chat_sender);
+      text_color = ContextCompat.getColor(rootView.getContext(), R.color.chat_sender_text);
       messageContainer.setGravity(Gravity.RIGHT);
+      userName.setVisibility(View.GONE);
     } else {
-      color = ContextCompat.getColor(rootView.getContext(), R.color.user_chat_background_color);
-      leftArrow.setVisibility(View.VISIBLE);
-      rightArrow.setVisibility(View.GONE);
+      bg_color = ContextCompat.getColor(rootView.getContext(), R.color.chat_sendee);
+      text_color = ContextCompat.getColor(rootView.getContext(), R.color.chat_sendee_text);
       messageContainer.setGravity(Gravity.LEFT);
     }
 
-    ((GradientDrawable) messageBox.getBackground()).setColor(color);
-    ((RotateDrawable) leftArrow.getBackground()).getDrawable()
-        .setColorFilter(color, PorterDuff.Mode.SRC);
-    ((RotateDrawable) rightArrow.getBackground()).getDrawable()
-        .setColorFilter(color, PorterDuff.Mode.SRC);
+    ((GradientDrawable) messageBox.getBackground()).setColor(bg_color);
+    messageText.setTextColor(text_color);
+  }
+
+  public void setTimestamp(Date date){
+    int hours = date.getHours();
+    int minutes = date.getMinutes();
+    timestampText.setText(hours + ":" + minutes);
   }
 
   public void setName(String name) {
```

Comment:
I think these might be better as not being methods - they're not used more than once